sdk/textanalytics/Azure.AI.TextAnalytics/src/OperationIdInformation.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+
+namespace Azure.AI.TextAnalytics
+{
+    internal class OperationIdInformation
+    {
+        // Parameterless constructor required for JSON deserialization.
+        public OperationIdInformation() { }
+
+        public OperationIdInformation(string jobId, IDictionary<string, int> indexMap, bool? showStats)
+        {
+            JobId = jobId;
+            InputDocumentOrder = new Dictionary<string, int>(indexMap);
+            ShowStats = showStats;
+        }
+
+        public Dictionary<string, int> InputDocumentOrder { get; set; }
+
+        public bool? ShowStats { get; set; }
+
+        public string JobId { get; set; }
+
+        public static OperationIdInformation Decode(string base64OperationId)
+        {
+            byte[] base64EncodedBytes = Convert.FromBase64String(base64OperationId);
+            string plainOperationId = Encoding.UTF8.GetString(base64EncodedBytes);
+
+            return JsonSerializer.Deserialize<OperationIdInformation>(plainOperationId);
```

Comment:
We should add validation here that all the required parts of `OperationIdInformation` are there and nothing else extra has appeared (unless we add a synthetic version - then we don't have to care as much about extra values).

Reply:
Also, is there a `ReadOnlySpan<byte>` overload of `Deserialize` we can use to avoid allocating an extra string?  It's not a major perf hit though if that's not easy.

Reply:
Added validation + version information in the `OperationIdInformation` class, similar to what we do in Search.

There is a `ReadOnlySpan<byte>` overload. Updated code to start using it.